Common/src/main/java/com/blamejared/crafttweaker/impl/command/type/script/example/ExamplesCommand.java

```diff
@@ -44,33 +45,37 @@ private static int execute(ServerPlayer player) {
             writeScriptFile(new ResourceManagerSourceFile(file, resourceManager));
         }
         
-        player.sendMessage(CommandUtilities.openingFile(new TranslatableComponent("crafttweaker.command.example.generated").withStyle(ChatFormatting.GREEN), getExamplesDir().getPath()), CraftTweakerConstants.CRAFTTWEAKER_UUID);
+        player.sendMessage(CommandUtilities.openingFile(new TranslatableComponent("crafttweaker.command.example.generated").withStyle(ChatFormatting.GREEN), getExamplesDir().toString()), CraftTweakerConstants.CRAFTTWEAKER_UUID);
         return Command.SINGLE_SUCCESS;
     }
     
-    private static void writeScriptFile(SourceFile sourceFile) {
+    private static void writeScriptFile(final SourceFile sourceFile) {
         
-        final File file = new File(getExamplesDir(), sourceFile.getFilename());
-        if(file.exists()) {
+        final Path file = getExamplesDir().resolve(sourceFile.getFilename());
+        if(Files.exists(file)) {
             CraftTweakerAPI.LOGGER.info("Skip writing example file '{}' since it already exists", file);
             return;
         }
         
-        if(!file.getParentFile().exists() && !file.getParentFile().mkdirs()) {
-            CraftTweakerAPI.LOGGER.error("Could not create folder '{}'", file.getParentFile());
-            return;
+        if(!Files.exists(file.getParent())) {
+            try {
+                Files.createDirectories(file.getParent());
+            } catch(final IOException e) {
+                CraftTweakerAPI.LOGGER.error("Could not create folder '" + file.getParent() + "'", e);
+                return;
+            }
         }
         
-        try(final FileWriter writer = new FileWriter(file, false); final Reader reader = sourceFile.open()) {
-            IOUtils.copy(reader, writer);
-        } catch(IOException e) {
+        try(final Reader reader = sourceFile.open()) {
+            Files.copy(IOUtils.toInputStream(IOUtils.toString(reader), StandardCharsets.UTF_8), file);
```

Comment:
Why from `Reader->string->InputStream->File`?
instead of bulding an inputStream based on the Reader?

Reply:
How?